third_party/amd/backend/compiler.py

```diff
@@ -129,12 +135,14 @@ def get_module_map(self) -> Dict[str, ModuleType]:
     def load_dialects(self, ctx):
         amd.load_dialects(ctx)
 
+    MAX_INT_32 = 2**31 - 1
+
     @staticmethod
     def is_within_2gb(arg):
         if hasattr(arg, "ptr_range"):
-            return arg.ptr_range() <= 2**31 - 1
-        if "torch.Tensor" in str(type(arg)) and hasattr(arg, "untyped_storage"):
-            return arg.untyped_storage().size() <= 2**31 - 1
+            return arg.ptr_range() <= HIPBackend.MAX_INT_32
+        if isinstance(arg, torch.Tensor) and hasattr(arg, "untyped_storage"):
```

Comment:
This string comparison was significantly slower than a generic isinstance check, which also much better Python style.

Reply:
This means we are pulling `torch` as a dependency for the compiler component, which haven't been the case thus far. (Runtime component is doing that, which make sense given we need `torch` for tensors/buffers.) Curious, how much slower is it? We actually thought about doing this way before; but decided to do string comparison due to dependency.

Reply:
Okay thanks for the metrics. With that I think I'm fine for now. Can you move the `import torch` at the beginning of the file into this function then? So it makes it clear we only need it for this function. And if `use_buffer_ops` is off we don't need to `import` at all given the condition at L163.